package/zmq_adapter/src/zmq_adapter_udp_connect.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Jacob McNamee <jacob@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include "zmq_adapter.h"
+
+#define CONNECT_RETRY_TIME_s 10
+
+static int addr_parse(const char *addr, struct sockaddr_in *s_addr)
+{
+  char ip[16];
+  int port;
+  if (sscanf(addr, "%15[^:]:%d", ip, &port) != 2) {
+    syslog(LOG_ERR, "error parsing address");
+    return 1;
+  }
+
+  memset(s_addr, 0, sizeof(*s_addr));
+  s_addr->sin_family = AF_INET;
+  s_addr->sin_port = htons(port);
+
+  if (inet_aton(ip, &s_addr->sin_addr) == 0) {
+    syslog(LOG_ERR, "invalid address");
+    return 1;
+  }
+
+  return 0;
+}
+
+static int socket_create(const struct sockaddr_in *addr)
+{
+  int ret;
+
+  int fd = socket(AF_INET, SOCK_DGRAM, 0);
+  if (fd < 0) {
+    return fd;
+  }
+
+  ret = connect(fd, (struct sockaddr *)addr, sizeof(*addr));
+  if (ret != 0) {
+    goto err;
+  }
+
+  return fd;
+
+err:
+  close(fd);
+  fd = -1;
+  return ret;
+}
+
+int udp_connect_loop(const char *addr)
+{
+  struct sockaddr_in s_addr;
+  if (addr_parse(addr, &s_addr) != 0) {
+    return 1;
+  }
+
+  while (1) {
+    int fd = socket_create(&s_addr);
+    if (fd < 0) {
+      debug_printf("error connecting UDP socket\n");
+      sleep(CONNECT_RETRY_TIME_s);
+      continue;
+    }
+
+    io_loop_start(-1, fd);
```

Comment:
@denniszollo @lbara @silverjam I think we should use this similarly in the TCP client to only receive data for the piksi and NOT send the piksi data out.

Reply:
Was going to say the same thing about the TCP server, but that's a bi-directional link :(